controller/tests/array_action/xiv/array_mediator_xiv_tests.py

```diff
@@ -116,58 +116,67 @@ def test_create_volume__generate_volume_response_raise_exception(self, response)
         with self.assertRaises(Exception):
             self.mediator.create_volume("volume", 10, None, "pool1")
 
-    def test_copy_to_existing_volume_from_snapshot_succeeds_with_resize(self):
+    def _test_copy_to_existing_volume_from_snapshot(self, src_snapshot_capacity_in_bytes,
+                                                    min_volume_size_in_bytes):
+        volume_id = "volume_id"
+        source_id = "source_id"
         volume_name = "volume"
         src_snapshot_name = "snapshot"
-        src_snapshot_capacity_in_bytes = 500
-        min_volume_size_in_bytes = 1000
         self.mediator.client.cmd.vol_format = Mock()
         self.mediator.client.cmd.vol_copy = Mock()
         self.mediator.client.cmd.vol_resize = Mock()
-        self.mediator.copy_to_existing_volume_from_source(volume_name, src_snapshot_name,
+        target_volume = self._get_cli_volume(name=volume_name)
+        source_volume = self._get_cli_volume(name=src_snapshot_name)
+        self.mediator.client.cmd.vol_list.side_effect = [Mock(as_single_element=target_volume),
+                                                         Mock(as_single_element=source_volume)]
+        self.mediator.copy_to_existing_volume_from_source(volume_id, source_id,
                                                           src_snapshot_capacity_in_bytes, min_volume_size_in_bytes)
-        volume_size_in_blocks = 1
+        calls = [call(wwn=volume_id), call(wwn=source_id)]
+        self.mediator.client.cmd.vol_list.assert_has_calls(calls, any_order=False)
```

Comment:
do we really want to enforce the order?

Reply:
yes, if it's in any other order so it means that something went wrong. that is the order of the callings in the [method](https://github.com/IBM/ibm-block-csi-driver/blob/57287fe734bc3c22f09e59cc890b34d306b41861/controller/array_action/array_mediator_xiv.py#L223-L224)

Reply:
why is it wrong for the method to call them in reversed order?

Reply:
how can you be sure that the right value returns to each parameter? (e.g. the call for the `source_name ` value should be called with `source_id`)

Reply:
got it